test/e2e/misc_test.go

```diff
@@ -76,6 +88,136 @@ func (suite *MiscTestSuite) TestValidateBuildImage() {
 	}
 }
 
+func (suite *MiscTestSuite) TestBasicOAuth() {
+	if !isOpenShift(t) {
+		t.Skip("This test only runs on Openshift")
+	}
+
+	username := "e2etestuser"
+	password := "befuddled"
+	// To update this, use the output from: htpasswd -nbs e2etestuser befuddled
+	userPasswordHash := "e2etestuser:{SHA}MfHY85GCR7WcTE7cQ2CGmXg9uTA="
+
+	userPasswordSecret := make(map[string][]byte)
+	userPasswordSecret["htpasswd"] = []byte(userPasswordHash)
+	secret := createSecret("test-oauth-secret", namespace, userPasswordSecret)
+
+	jaegerInstanceName := "test-oauth"
+	jaeger := jaegerWithPassword(namespace, jaegerInstanceName, secret.Name)
+	err := fw.Client.Create(context.Background(), jaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying jaeger")
+	defer undeployJaegerInstance(jaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for Jaeger deployment")
+
+	urlPattern := "%s/api/services"
+	route := findRoute(t, fw, jaegerInstanceName)
+	require.Len(t, route.Status.Ingress, 1, "Wrong number of ingresses.")
+	url := fmt.Sprintf("https://"+urlPattern, route.Spec.Host)
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	httpClient := http.Client{Timeout: 30 * time.Second, Transport: transport}
+	log.Infof("Using Query URL [%v]", url)
+
+	// A request without credentials should return a 403
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	require.NoError(t, err, "Failed to create httpRequest")
+	response, err := httpClient.Do(request)
+	require.Equal(t, response.StatusCode, 403)
+
+	// Add basic auth to the request and retry
+	request.SetBasicAuth(username, password)
+	response, err = httpClient.Do(request)
+	body, err := ioutil.ReadAll(response.Body)
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
Why is the request being repeated before and inside the poll? Is it that the first request is causing the `jaeger-query` service to be added to the list?

Reply:
Yes, this was to make sure there was at least one trace.  Since there's the wait loop it's redundant, so I've removed it.